mooncake-integration/store/store_py.cpp

```diff
@@ -725,6 +725,158 @@ int DistributedObjectStore::get_into(const std::string &key, void *buffer,
+    const std::vector<size_t> &sizes) {
+    auto start_time = std::chrono::steady_clock::now();
+    if (!client_) {
+        LOG(ERROR) << "Client is not initialized";
+        return std::vector<int>(keys.size(), -1);
+    }
+
+    if (keys.size() != buffers.size() || keys.size() != sizes.size()) {
+        LOG(ERROR) << "Mismatched sizes for keys, buffers, and sizes";
+        return std::vector<int>(keys.size(), -1);
+    }
+
+    std::vector<int> results(keys.size());
+    mooncake::Client::BatchObjectInfo
+        object_infos;  // This is BatchGetReplicaListResponse
+
+    // Step 1: Batch query object info
+    ErrorCode batch_query_err = client_->BatchQuery(keys, object_infos);
+    if (batch_query_err != ErrorCode::OK) {
+        LOG(ERROR) << "BatchQuery failed with error: "
+                   << toString(batch_query_err);
+        std::fill(results.begin(), results.end(), toInt(batch_query_err));
+        return results;
+    }
+
+    // Step 2: Prepare slices for each key
+    std::unordered_map<std::string, std::vector<mooncake::Slice>> all_slices;
+    std::vector<std::string> valid_keys;
+    for (size_t i = 0; i < keys.size(); ++i) {
+        const auto &key = keys[i];
+        auto it = object_infos.batch_replica_list.find(key);
+
+        if (it == object_infos.batch_replica_list.end()) {
+            results[i] = -toInt(ErrorCode::OBJECT_NOT_FOUND);
+            continue;
+        }
+
+        auto &replica_list = it->second;
+        if (replica_list.empty()) {
+            LOG(ERROR) << "Internal error: replica_list is empty for key: "
+                       << key;
+            results[i] = -1;
+            continue;
+        }
+
+        auto &replica = replica_list[0];
+        uint64_t total_size = 0;
+        for (auto &handle : replica.buffer_descriptors) {
+            total_size += handle.size_;
+        }
+
+        if (sizes[i] < total_size) {
+            LOG(ERROR) << "User buffer too small for key: " << key
+                       << ". Required: " << total_size
+                       << ", provided: " << sizes[i];
+            results[i] = -1;
+            continue;
+        }
+
+        uint64_t offset = 0;
+        std::vector<mooncake::Slice> key_slices;
+        for (auto &handle : replica.buffer_descriptors) {
+            void *chunk_ptr = static_cast<char *>(buffers[i]) + offset;
+            key_slices.emplace_back(Slice{chunk_ptr, handle.size_});
+            offset += handle.size_;
+        }
+        all_slices[key] = key_slices;
+        results[i] = static_cast<int>(total_size);
+        valid_keys.push_back(key);
+    }
+
+    if (valid_keys.empty()) {
+        return results;
+    }
+
+    // Step 3: Batch get data
+    ErrorCode batch_get_err =
+        client_->BatchGet(valid_keys, object_infos, all_slices);
+    if (batch_get_err != ErrorCode::OK) {
+        LOG(ERROR) << "BatchGet failed with error: " << toString(batch_get_err);
+        for (const auto &key : valid_keys) {
```

Comment:
Finding key for all keys is O(N^2) time complexity. Since N is usually larger than 100, do we have a better solution?

Reply:
After #562 , I will resolve it.